From df431b16f0f870ac410767851dc9c96f38761aa6 Mon Sep 17 00:00:00 2001
From: Takahiro Terada <takahiro.terada@miraclelinux.com>
Date: Wed, 10 Apr 2024 11:49:03 +0900
Subject: [PATCH] Lib/test/test_zipfile.py: Fix test_zipfile test crash

When executing the test_zipfile, following error occurs because debian's
"0001-3.9-gh-109858-Protect-zipfile-from-quoted-overlap-zi.patch" file was
mistake.

  # python3 -m test -v test_zipfile
  ...
  test test_zipfile crashed -- Traceback (most recent call last):
    File "/usr/lib/python3.7/test/libregrtest/runtest.py", line 166, in runtest_inner
      the_module = importlib.import_module(abstest)
    File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
    File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
    File "<frozen importlib._bootstrap>", line 983, in _find_and_load
    File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
    File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
    File "<frozen importlib._bootstrap_external>", line 728, in exec_module
    File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    File "/usr/lib/python3.7/test/test_zipfile.py", line 1210, in <module>
      class OtherTests(unittest.TestCase):
    File "/usr/lib/python3.7/test/test_zipfile.py", line 1683, in OtherTests
      @requires_zlib()
  TypeError: _id() missing 1 required positional argument: 'obj'

  test_zipfile failed
  ...

The debian's "0001-3.9-gh-109858-Protect-zipfile-from-quoted-overlap-zi.patch"
file was backported from the added code for Python 3.9.
The notation of decorator "@requires_zlib()" is correct in Python 3.9, but not
in Python 3.7.

So, fix notation of decorator for Python 3.7.

Signed-off-by: Takahiro Terada <takahiro.terada@miraclelinux.com>
---
 Lib/test/test_zipfile.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Lib/test/test_zipfile.py b/Lib/test/test_zipfile.py
index a9ce311..d54317a 100644
--- a/Lib/test/test_zipfile.py
+++ b/Lib/test/test_zipfile.py
@@ -1680,7 +1680,7 @@ class OtherTests(unittest.TestCase):
                 fp.seek(0, os.SEEK_SET)
                 self.assertEqual(fp.tell(), 0)
 
-    @requires_zlib()
+    @requires_zlib
     def test_full_overlap(self):
         data = (
             b'PK\x03\x04\x14\x00\x00\x00\x08\x00\xa0lH\x05\xe2\x1e'
@@ -1709,7 +1709,7 @@ class OtherTests(unittest.TestCase):
             with self.assertRaisesRegex(zipfile.BadZipFile, 'File name.*differ'):
                 zipf.read('b')
 
-    @requires_zlib()
+    @requires_zlib
     def test_quoted_overlap(self):
         data = (
             b'PK\x03\x04\x14\x00\x00\x00\x08\x00\xa0lH\x05Y\xfc'
-- 
2.25.1

